processor/src/test/java/org/mapstruct/ap/test/injectionstrategy/dagger2/constructor/Dagger2ConstructorMapperTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.injectionstrategy.dagger2.constructor;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mapstruct.ap.test.injectionstrategy.shared.CustomerDto;
+import org.mapstruct.ap.test.injectionstrategy.shared.CustomerEntity;
+import org.mapstruct.ap.test.injectionstrategy.shared.Gender;
+import org.mapstruct.ap.test.injectionstrategy.shared.GenderDto;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.GeneratedSource;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+import static java.lang.System.lineSeparator;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test constructor injection for component model dagger2.
+ */
+@WithClasses({
+    CustomerDto.class,
+    CustomerEntity.class,
+    Gender.class,
+    GenderDto.class,
+    CustomerDagger2ConstructorMapper.class,
+    GenderDagger2ConstructorMapper.class,
+    ConstructorDagger2Config.class
+})
+@ComponentScan(basePackageClasses = CustomerDagger2ConstructorMapper.class)
+@Configuration
```

Comment:
Why are we using Spring for this?

Reply:
These tests were basically copy and pasted from [the JRS330 tests](https://github.com/mapstruct/mapstruct/blob/master/processor/src/test/java/org/mapstruct/ap/test/injectionstrategy/jsr330/constructor/Jsr330ConstructorMapperTest.java), with updated checks for Dagger behavior

Reply:
Interesting. I think that the JSR 330 tests are like that because Spring supports that as well. Makes sense to only check the generated code here. 

The integration tests are testing the full integration from what I can see, so that's great